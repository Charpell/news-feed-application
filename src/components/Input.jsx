import React from 'react';

/**
 * Render an input field of text
 * @returns {inputField}
 */
function Input(props) {
  const { search, onSearch } = props;
  return (
    <input
      className="form-control"
      placeholder="Search for your favourite news sources on the go..."
      type="text"
      value={search}
      onChange={onSearch}
    />
  );
}

export default Input;
