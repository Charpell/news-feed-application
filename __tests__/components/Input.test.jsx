import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../../src/components/Input';

describe('Input', () => {
  it('renders a snapshot', () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
