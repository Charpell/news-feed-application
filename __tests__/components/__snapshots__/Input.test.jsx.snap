// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders a snapshot 1`] = `
<input
  className="form-control"
  onChange={undefined}
  placeholder="Search for your favourite news sources on the go..."
  type="text"
  value={undefined}
/>
`;
